### PR TITLE
Add documentation for `.govuk-input--extra-letter-spacing`

### LINKED ISSUE #2665: Add/update documentation for new `.govuk-input--extra-letter-spacing` class
## Related documentation
Directly affects: 
- [Text input](https://design-system.service.gov.uk/components/text-input/) component

Probably affects:
- [Bank details](https://design-system.service.gov.uk/patterns/bank-details/) pattern
- [National Insurance numbers](https://design-system.service.gov.uk/patterns/national-insurance-numbers/) pattern
- [Confirm a phone number](https://design-system.service.gov.uk/patterns/confirm-a-phone-number/) pattern

May affect:
- [Payment card details](https://design-system.service.gov.uk/patterns/payment-card-details/) pattern
- [Telephone numbers](https://design-system.service.gov.uk/patterns/telephone-numbers/) pattern

## Suggestion
The next version of GOV.UK Frontend, which is currently intended to be the last feature release for v4, has introduced a new class for text inputs: `.govuk-input--extra-letter-spacing`

This class intends to provide easier to read formatting for form inputs requiring numeric and alphanumeric strings. It does this by increasing the space between each character and by enabling [tabular number styling](https://www.fonts.com/content/learning/fontology/level-3/numbers/proportional-vs-tabular-figures), which modifies how numbers are spaced out and changes the appearance of the number 1 to be wider and more visually distinct. 

This is intended for "non-word" inputs that a user is unlikely to have memorised and may benefit from being able to easily distinguish each individual character, such as:

- Security and OTP codes
- National Insurance numbers
- Bank card numbers, sort codes, BIC and IBAN codes
- Reference and application numbers
- (Potentially) telephone numbers

We should consider documenting this new class and the intended use cases for it. We should also update any examples of the above use cases to include it.

### PR DESCRIPTION
**Pull request by querkmachine**
Preparing some documentation for one of the new features coming in the next release of GOV.UK Frontend. 

> **Note**: I've updated the code examples with the new class name, but the changes won't actually happen until the new version of Frontend has been released and the Design System has been updated to use it. 

Closes #2665.

## Changes
- Adds a section to the Text Input component docs explaining when and how to use the new class.
- Adds an example of the Text Input component using the new class, using a 'company authentication code' from Companies House as an example.
- Updates examples on related patterns to use the new class. 

## Thoughts
- Is the company authentication code example understandable? 
  - I was trying to use something other than 2FA code, NI number or bank details, as we already have examples of these on other pages.
  - Disclaimer: I'm not sure whether the hint text provided is actually accurate. 
  - Company authentication codes are six characters long and composed of a mixture of uppercase letters and digits, by the way.
- Should the example have the input's value pre-populated? 
  - Given the function of the class is to change how the value is presented, having this pre-populated saves a reader having to type to see the differences. 
  - One of the most striking differences (when using the Transport font) is the visual difference between the proportional and tabular versions of '1'. A pre-populated example could make this more obvious.

### CODE REVIEW

File: src/components/text-input/index.md [hunk 1]
```diff
@@ -116,6 +116,20 @@ Do not set the `inputmode` attribute to `decimal` as it causes some devices to b
 
 Do not use `<input type="number">` unless your user research shows that there’s a need for it. With `<input type="number">` there’s a risk of users accidentally incrementing a number when they’re trying to do something else - for example, scroll up or down the page. And if the user tries to enter something that’s not a number, there’s no explicit feedback about what they’re doing wrong.
 
+### Codes and sequences
+
+If you're asking the user to enter a code or sequence that they're unlikely to have memorised - such as an application reference ID, account number or security code - style the input's text to visually separate each character. This makes it easier for the user to visually check that the code they've typed is correct.
```
Comment (claireashworth) [inline src/components/text-input/index.md:121]:
```suggestion
To make it easier for the user to visually check that the code they've typed is correct, you must style the input's text to visually separate each character. This is important if you're asking the user to enter a code or sequence they're unlikely to have memorised, such as an application reference ID, account number or security code. 
```

File: src/components/text-input/index.md [hunk 2]
```diff
@@ -116,6 +116,20 @@ Do not set the `inputmode` attribute to `decimal` as it causes some devices to b
...
+
+You don't need to do this for memorable information, like phone numbers and postcodes.
```
Comment (claireashworth) [inline src/components/text-input/index.md:123]:
```suggestion
You do not need to do this for memorable information, like phone numbers and postcodes.
```

File: src/components/text-input/index.md (hunk 1 above)
Comment (calvin-lau-sig7) [reply src/components/text-input/index.md:121]: Yeah, I like the approach. For the first sentence, we could match the a bit of a pattern this page has developed and say:

> Help the user visually check that the code they've typed is correct by styling the input's text to separate each character.

or 

> Style the input's text to separate each character to help the user visually check the code they've typed in is correct.

Comment (claireashworth) [reply src/components/text-input/index.md:121]: @calvin-lau-sig7 I've amended my suggestion with your first option.

File: src/components/text-input/index.md [hunk 3]
```diff
@@ -116,6 +116,20 @@ Do not set the `inputmode` attribute to `decimal` as it causes some devices to b
 
 Do not use `<input type="number">` unless your user research shows that there’s a need for it. With `<input type="number">` there’s a risk of users accidentally incrementing a number when they’re trying to do something else - for example, scroll up or down the page. And if the user tries to enter something that’s not a number, there’s no explicit feedback about what they’re doing wrong.
 
+### Codes and sequences
+
+Help the user visually check that the code they've typed is correct by styling the input's text to visually separate each character. This is important if you're asking the user to enter a code or sequence they're unlikely to have memorised, such as an application reference ID, account number or security code. 
+
+You do not need to do this for memorable information, like phone numbers and postcodes.
```
Comment (claireashworth) [inline src/components/text-input/index.md:123]:
```suggestion
You do not need to do this for memorable information, such as phone numbers and postcodes.
```

**Review by claireashworth — APPROVED** [review]

**Pull request merged**